WordPress/src/main/java/org/wordpress/android/localcontentmigration/LocalMigrationContentProvider.kt

```diff
@@ -0,0 +1,82 @@
+package org.wordpress.android.localcontentmigration
+
+import android.database.Cursor
+import android.net.Uri
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.localcontentmigration.LocalContentEntity.EligibilityStatus
+import org.wordpress.android.localcontentmigration.LocalContentEntity.Post
+import org.wordpress.android.localcontentmigration.LocalContentEntity.Site
+import org.wordpress.android.provider.query.QueryResult
+import java.lang.Integer.parseInt
+
+class LocalMigrationContentProvider: TrustedQueryContentProvider() {
+    @EntryPoint
+    @InstallIn(SingletonComponent::class)
+    interface LocalMigrationContentProviderEntryPoint {
+        fun queryResult(): QueryResult
+        fun localSiteProviderHelper(): LocalSiteProviderHelper
+        fun localPostProviderHelper(): LocalPostProviderHelper
+        fun localEligibilityStatusProviderHelper(): LocalEligibilityStatusProviderHelper
+    }
+
+    override fun query(uri: Uri): Cursor {
+        val path = uri.path!!
+        val (entity, groups) = LocalContentEntity.values().firstNotNullOf { entity ->
+            entity.contentIdCapturePattern.find(path)?.let { match ->
+                return@firstNotNullOf Pair(entity, match.groups)
+            }
+        }
+        val parameters = groups.drop(1).mapNotNull { it?.value?.let { id -> parseInt(id) } }
```

Comment:
Would it make sense to extract this parameters mapping logic that is calling `drop(1)` and `getOrNull` into a separate method with a meaningful name? I'm thinking that it might help with maintainability.

Reply:
Good idea! I meant to at least leave a comment, but a private function would be even better. Thanks for the suggestion :+1:

Reply:
Updated here: https://github.com/wordpress-mobile/WordPress-Android/pull/17398/commits/302940fc5c2f3288b90507fbb5eaab537b15199b